### ISSUE TITLE
[rspec after suite] stream closed (IOError) on Ruby 2.3.4

### ISSUE THREAD

**Issue by olleolleolle** [bundler/gemstash#184]
### Subject of the issue

stream closed (IOError)

Similar to https://github.com/rspec/rspec-core/issues/1541
which was fixed in https://github.com/rspec/rspec-core/pull/1544/

Reported in https://github.com/bundler/gemstash/pull/183

### Your environment

* Ruby version: 2.3.4
* rspec-core version: 3.7.1

### Steps to reproduce

"Run the Gemstash test suite on Ruby 2.3.4."

Example build failure: https://travis-ci.org/bundler/gemstash/jobs/387726953#L1270

### Expected behavior

No error report at the end of `rspec`.

### Actual behavior

A stacktrace at the end of `rspec`:

```
/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/set.rb:33:in `each': stream closed (IOError)
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/set.rb:33:in `each'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:58:in `to_a'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:58:in `registered_listeners'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:205:in `notify'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:120:in `example_group_finished'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/example_group.rb:602:in `run'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `block (3 levels) in run_specs'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `map'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:118:in `block (2 levels) in run_specs'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/configuration.rb:1926:in `with_suite_hooks'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:113:in `block in run_specs'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/reporter.rb:79:in `report'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:112:in `run_specs'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:87:in `run'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:71:in `run'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib/rspec/core/runner.rb:45:in `invoke'
	from /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/exe/rspec:4:in `<main>'
/home/travis/.rvm/rubies/ruby-2.3.4/bin/ruby -I/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/lib:/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-support-3.7.1/lib /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.7.1/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb --color failed
```

**Comment by olleolleolle** [comment]
[Last build on master, this time on Ruby 2.4.1](https://travis-ci.org/bundler/gemstash/jobs/394312107) had this problem, too.

```
/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.4.0/gems/rspec-core-3.7.1/exe/rspec: stream closed (IOError)
/home/travis/.rvm/rubies/ruby-2.4.1/bin/ruby -I/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.4.0/gems/rspec-core-3.7.1/lib:/home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.4.0/gems/rspec-support-3.7.1/lib /home/travis/build/bundler/gemstash/vendor/bundle/ruby/2.4.0/gems/rspec-core-3.7.1/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb --color failed
The command "bundle exec rake" exited with 1.
```

**Comment by olleolleolle** [comment]
Today, Ruby 2.6.0 was released, and we're nearing March 2019, when 2.3.x will be retired:

https://www.ruby-lang.org/en/news/2018/10/17/ruby-2-3-8-released/

This Issue is unlikely to get any more attention. I will close it.
